src/api/util/url-add.js

```diff
@@ -22,28 +22,41 @@ module.exports = (send) => {
       opts = {}
     }
 
-    if (typeof url !== 'string' ||
-        !url.startsWith('http')) {
+    if (!validUrl(url)) {
       return callback(new Error('"url" param must be an http(s) url'))
     }
 
     callback = once(callback)
 
-    request(parseUrl(url).protocol)(url, (res) => {
-      res.once('error', callback)
-      if (res.statusCode >= 400) {
-        return callback(new Error(`Failed to download with ${res.statusCode}`))
-      }
+    requestWithRedirect(url, opts, send, callback)
+  })
+}
+
+const validUrl = (url) => typeof url === 'string' && url.startsWith('http')
+
+const requestWithRedirect = (url, opts, send, callback) => {
+  request(parseUrl(url).protocol)(url, (res) => {
+    res.once('error', callback)
+    if (res.statusCode >= 400) {
+      return callback(new Error(`Failed to download with ${res.statusCode}`))
+    }
 
+    const redirection = res.headers.location
+
+    if (redirection) {
```

Comment:
Which header is that if not `location`? That's the header in the [spec](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.3) and that seem to be [used by the request module too](https://github.com/request/request/blob/f009af231412fe8fee4ab5dd24d24781fd731fc3/lib/redirect.js#L50).